multicluster/controllers/multicluster/leader/labelidentity_export_controller_test.go

```diff
@@ -185,6 +191,61 @@ func TestLabelIdentityResourceExportReconclie(t *testing.T) {
 	}
 }
 
+func TestConcurrentProcessLabelForResourceImport(t *testing.T) {
+	existingResImp := &mcsv1alpha1.ResourceImport{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: common.LeaderNamespace,
+			Name:      labelHash3,
+		},
+		Spec: mcsv1alpha1.ResourceImportSpec{
+			ClusterIDs: []string{common.LocalClusterID},
+			LabelIdentity: &mcsv1alpha1.LabelIdentitySpec{
+				Label: normalizedLabel3,
+				ID:    3,
+			},
+		},
+	}
+	existResImpList := &mcsv1alpha1.ResourceImportList{
+		Items: []mcsv1alpha1.ResourceImport{*labelIdentityResImp, *existingResImp},
+	}
+
+	stopCh := make(chan struct{})
+	defer close(stopCh)
+	fakeClient := fake.NewClientBuilder().WithScheme(common.TestScheme).WithLists(existResImpList).Build()
+	r := NewLabelIdentityExportReconciler(fakeClient, common.TestScheme, common.LeaderNamespace)
+	r.hashToLabels = map[string]string{
+		labelHash:  normalizedLabel,
+		labelHash2: normalizedLabel2,
+	}
+	// Mock the state of the reconciler where id 3 was allocated for labelHash3
+	r.labelsToID.Store(labelHash3, uint32(3))
+	r.allocator.setAllocated(3)
+	// The three keys should be drawn out of the queue almost concurrently after the worker starts
+	r.labelQueue.Add(labelHash)
+	r.labelQueue.Add(labelHash2)
+	r.labelQueue.Add(labelHash3)
```

Comment:
I am wondering does this simulate the performance stress with 3 keys? Have you tried this test without changing the type of labelsToID? 
I suppose you are trying to fix the issue that label Identity resource import is not created with multiple Pod label changes? Could you explain more why concurrent map writes by multiple workers of the label queue will cause it?
Better to add the issue symptom in the PR description as well.

Reply:
I met this issue while doing a performance test. Using a normal map will cause panic: `fatal error: concurrent map writes`.

Reply:
I modified the test a bit to simulate more events and more workers concurrently working on the label queue. Without the fix, I can consistently hit concurrent map write issue:
```
=== RUN   TestConcurrentProcessLabelForResourceImport
fatal error: concurrent map writes

goroutine 110 [running]:
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).handleLabelIdentityDelete(0x14000794180, {0x10402a650, 0x14000048110}, {0x14000043f50, 0x10})
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:246 +0x18c
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).syncLabelResourceImport(0x14000794180, {0x14000043f50, 0x10})
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:222 +0x174
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).processLabelForResourceImport(0x14000794180)
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:195 +0xb8
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).labelQueueWorker(0x1400006ee88?)
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:181 +0x28
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x0?)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:155 +0x40
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x104016c40, 0x14000402330}, 0x1, 0x1400016c180)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:156 +0x90
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:133 +0x80
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:90 +0x28
created by antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).Run
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:175 +0x6c

goroutine 105 [runnable]:
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).handleLabelIdentityAdd(0x14000794180, {0x10402a650, 0x14000048110}, {0x14000042450, 0x10}, {0x103b7d500, 0x35})
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:256 +0x88
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).syncLabelResourceImport(0x14000794180, {0x14000042450, 0x10})
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:216 +0x14c
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).processLabelForResourceImport(0x14000794180)
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:195 +0xb8
antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).labelQueueWorker(0x14000506688?)
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:181 +0x28
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0x30000001000000?)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:155 +0x40
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0x0?, {0x104016c40, 0x140004627b0}, 0x1, 0x1400016c180)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:156 +0x90
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0x0?, 0x3b9aca00, 0x0, 0x0?, 0x0?)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:133 +0x80
k8s.io/apimachinery/pkg/util/wait.Until(0x0?, 0x0?, 0x0?)
	/Users/dingyang/go/pkg/mod/k8s.io/apimachinery@v0.24.9/pkg/util/wait/wait.go:90 +0x28
created by antrea.io/antrea/multicluster/controllers/multicluster/leader.(*LabelIdentityExportReconciler).Run
	/Users/dingyang/go/src/github.com/antrea-io/antrea/multicluster/controllers/multicluster/leader/labelidentity_export_controller.go:175 +0x6c
```